homeassistant/components/stream/worker.py

```diff
@@ -27,31 +31,35 @@
 class SegmentBuffer:
     """Buffer for writing a sequence of packets to the output as a segment."""
 
-    def __init__(self, outputs_callback, ll_hls) -> None:
+    def __init__(
+        self, outputs_callback: Callable[[], Mapping[str, StreamOutput]], ll_hls: bool
+    ) -> None:
         """Initialize SegmentBuffer."""
-        self._stream_id = 0
-        self._outputs_callback = outputs_callback
+        self._stream_id: int = 0
+        self._outputs_callback: Callable[
+            [], Mapping[str, StreamOutput]
+        ] = outputs_callback
         # Each element is a StreamOutput
-        self._outputs: list[StreamOutput] = []
-        self._sequence = -1
-        self._segment_start_pts = None
-        self._memory_file = None
-        self._av_output = None
-        self._segment = None
-        self._input_video_stream = None
+        self._outputs: ValuesView[StreamOutput] | list[StreamOutput] = []
+        self._sequence: int = -1
+        self._segment_start_pts: int
```

Comment:
I think it's dangerous to have undefined attributes in the class. Please assign some default value.

Reply:
I had them assigned to None before, but the problem is the type then becomes an optional type like int | None. Everything does get initialized later outside of __init__, but since the types are set in __init__, this ends up causing type checking errors later that require meaningless (wasteful?) assertions or casts to rectify.
Is there a good pattern to use to avoid this?

Reply:
A workaround for this can be defining default values with the same type, e.g. `-1`, `False`, empty `BytesIO`. This way it won't crash at least on access before initialisation of these attributes.

Reply:
Just going to use `x = cast(T, None)  # type: T` as shown in typing issue 81

Reply:
Sorry, I didn't see your comment above until just now. I didn't mean to just disregard your suggestion with my previous comment. But I would still prefer to have something undefined.
I suppose that with these simple types there is very little overhead to creating a new object that will just be wasted. However, even without the overhead, the thinking is I am sure I am initializing everything so I don't want to worry about assertions and Optional types, but even if I mess something up, I would prefer it to crash rather than to use an incorrect value. Does that make sense? (It's kind of moot here, but just wondering)

Reply:
Well, if you want it to crash then I think `assert is not None` is the way to go 🙂 This way you'll have proper type safety and it will crash in controlled places rather than random.

Reply:
Yes, I think that makes sense, but it does not really do a type promotion on the instance variables. So it seems like you would have to do an assertion in several places. I think type checking is useful and it does ensure safety even in this case, but I think the workarounds for this case (instance variables initialized outside of __init__) are not ideal and are actually quite a burden while being of limited benefit.
Anyway, we can address this later - as mentioned type hinting is really not the crux of this PR. :-)